jiant/models.py

```diff
@@ -1003,15 +1006,24 @@ def _seq_gen_forward(self, batch, task, predict):
 
         decoder = getattr(self, "%s_decoder" % task.name)
         out.update(decoder.forward(sent, sent_mask, batch["targs"]))
-        task.scorer1(out["loss"].item())
+        # Loss is not computed during generation/validation.
+        if "loss" in out:
+            task.scorer1(out["loss"].item())
 
         if "targs" in batch:
             # logits: batch_size * seq_len * tgt_voc_size
             target = batch["targs"]["words"][:, 1:].contiguous()
             target_mask = out["target_mask"]
-            logits = out["logits"]
-            task.update_metrics(logits, target, target_mask[:, 1:].contiguous())
 
+            assert "logits" in out or "predictions" in out
+
+            if "logits" in out:
+                logits = out["logits"]
+                task.update_metrics(logits, target, target_mask[:, 1:].contiguous())
```

Comment:
slight preference for using keyword args to make sure the order of args is right, e.g. `(logits=logits, mask=target_mask...)` and `(target=target, ...)`

Reply:
Done.